pkg/storage/s3/s3.go

```diff
@@ -111,13 +113,26 @@ func GetConfig(listers *regopclient.Listers) (*S3, error) {
 		}
 	}
 
+	cloudConfig, err := listers.OpenShiftConfigManaged.Get(defaults.KubeCloudConfigName)
+	switch {
+	case errors.IsNotFound(err):
+		// No cloud config, so no custom CA bundle.
+	case err != nil:
+		return nil, fmt.Errorf("unable to get the kube cloud config: %w", err)
+	default:
+		caBundle, ok := cloudConfig.Data[defaults.CloudCABundleKey]
+		if ok {
+			cfg.CABundle = caBundle
+		}
+	}
+
 	return cfg, nil
 }
 
-func (d *driver) updateConfigAndGetCredentials() (accessKey, secretKey string, err error) {
+func (d *driver) updateConfigAndGetCredentials() (*S3, error) {
```

Comment:
This function shouldn't return `*S3`. It makes really tempting to use it instead of `d.Config`.
If you don't want it to be a part of `spec.storage.s3` nor a part of the configmap `image-registry-private-configuration-user`, then use shouldn't touch this function.

Reply:
Either the config should have the name of the configmap and this name should be used only in getS3Service, or getS3Service should use the well-known configmap.

Reply:
Thanks. I have separated the code for fetching the custom CA bundle into its own function and removed it from `spec.storage.s3`.